From 644a6afaef4c4e727256d0d1df59d0fd602ddba0 Mon Sep 17 00:00:00 2001
From: Kathiravan Thirumoorthy <quic_kathirav@quicinc.com>
Date: Tue, 4 Jun 2024 16:13:20 +0530
Subject: [PATCH] arm64: disable DMA_BOUNCE_UNALIGNED_KMALLOC for ARCH_QCOM

Enabling DMA_BOUNCE_UNALIGNED_KMALLOC reserves additional memory for
the DMA bounce buffers, which is not needed for our use cases.

Disable it to save considerable amount of RAM.

Change-Id: Ice2240f30508d2315a57445db2649c1edf9b1914
Signed-off-by: Kathiravan Thirumoorthy <quic_kathirav@quicinc.com>
---
 arch/arm64/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/Kconfig b/arch/arm64/Kconfig
index 7adf2f0d4f13..4bdad8d7e929 100644
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -122,7 +122,7 @@ config ARM64
 	select CRC32
 	select DCACHE_WORD_ACCESS
 	select DYNAMIC_FTRACE if FUNCTION_TRACER
-	select DMA_BOUNCE_UNALIGNED_KMALLOC
+	select DMA_BOUNCE_UNALIGNED_KMALLOC if !ARCH_QCOM
 	select DMA_DIRECT_REMAP
 	select EDAC_SUPPORT
 	select FRAME_POINTER
-- 
2.34.1

